sysvabi64/sysvabi64.rst

```diff
@@ -1592,6 +1592,163 @@ IFUNC requirements for dynamic linkers
+    | FEAT_DIT          | 1U << 18 |
+    +-------------------+----------+
+    | FEAT_DPB          | 1U << 19 |
+    +-------------------+----------+
+    | FEAT_DPB2         | 1U << 20 |
+    +-------------------+----------+
+    | FEAT_JSCVT        | 1U << 21 |
+    +-------------------+----------+
+    | FEAT_FCMA         | 1U << 22 |
+    +-------------------+----------+
+    | FEAT_RCPC         | 1U << 23 |
+    +-------------------+----------+
+    | FEAT_RCPC2        | 1U << 24 |
+    +-------------------+----------+
+    | FEAT_FRINTTS      | 1U << 25 |
+    +-------------------+----------+
+    | FEAT_I8MM         | 1U << 27 |
+    +-------------------+----------+
+    | FEAT_BF16         | 1U << 28 |
+    +-------------------+----------+
+    | FEAT_SVE          | 1U << 31 |
+    +-------------------+----------+
+    | FEAT_SVE_F32MM    | 1U << 35 |
+    +-------------------+----------+
+    | FEAT_SVE_F64MM    | 1U << 36 |
+    +-------------------+----------+
+    | FEAT_SVE2         | 1U << 37 |
+    +-------------------+----------+
+    | FEAT_SVE_PMULL128 | 1U << 39 |
+    +-------------------+----------+
+    | FEAT_SVE_BITPERM  | 1U << 40 |
+    +-------------------+----------+
+    | FEAT_SVE_SHA3     | 1U << 41 |
+    +-------------------+----------+
+    | FEAT_SVE_SM4      | 1U << 42 |
+    +-------------------+----------+
+    | FEAT_SME          | 1U << 43 |
+    +-------------------+----------+
+    | FEAT_MEMTAG2      | 1U << 45 |
+    +-------------------+----------+
+    | FEAT_SB           | 1U << 47 |
+    +-------------------+----------+
+    | FEAT_SSBS2        | 1U << 50 |
+    +-------------------+----------+
+    | FEAT_BTI          | 1U << 51 |
+    +-------------------+----------+
+    | FEAT_WFXT         | 1U << 55 |
+    +-------------------+----------+
+    | FEAT_SME_F64      | 1U << 56 |
+    +-------------------+----------+
+    | FEAT_SME_I64      | 1U << 57 |
+    +-------------------+----------+
+    | FEAT_SME2         | 1U << 58 |
+    +-------------------+----------+
+    | FEAT_RCPC3        | 1U << 59 |
+    +-------------------+----------+
+    | FEAT_MOPS         | 1U << 60 |
+    +-------------------+----------+
+
+A special value ``FEAT_INIT = (1U << 63);`` is used to signify
+initialization completion.
+
+Implementing FMV using ``__aarch64_cpu_features`` is not required.
+Accessing ``__aarch64_cpu_features`` from outside a FMV resolver
+function is not well defined.
+
+Initialization of the ``__aarch64_cpu_features`` is performed by
+the runtime library. Each FMV resolver function will call the
+``__init_cpu_features_resolver`` function at load time. This runtime
+function must ensure that ``__aarch64_cpu_features`` is populated.
+
+Systems without support for IFUNC may call the
+``__init_cpu_features_resolver`` function from an
+``__attribute__((constructor))`` function provided by the runtime.
+
+Systems with support for IFUNC should place the global variable
```

Comment:
Just a thought. If ``__aarch64_cpu_features`` is made RELRO then if called as a constructor then ``__init_cpu_features_resolver`` must not write to it. 

Ideally if ``__aarch64_cpu_features`` is RELRO then we shouldn't also use the constructor, and if we do then all ``__init_cpu_features_resolver`` can do is check that ``__aarch64_cpu_features`` has already been initialised and return.